torch/csrc/jit/script/compiler.cpp

```diff
@@ -29,6 +29,113 @@ using ValueTable = std::unordered_map<std::string, SugaredValuePtr>;
 using AttributeMap = std::unordered_map<std::string, Const>;
 using ListAttributeMap = std::unordered_map<std::string, std::vector<Const>>;
 
+struct Refinements {
+
+  //using ordered map for deterministic graph output
+  std::map<std::string, TypePtr> types;
+  std::map<std::string, const SourceRange*> ranges;
+
+  // return the intersection of the values to type mappings in a and b whose
+  // types can be unified
+  static Refinements intersectRefinements(const Refinements& a, const Refinements& b) {
```

Comment:
Do you think it would be simpler to phrase these as in-place non-static methods? e.g. `a.intersect(b)` instead of `Refinements::interesctRefinements(a, b)`